presto-resource-group-managers/src/test/java/io/prestosql/plugin/resourcegroups/WithManagerSpec.java

```diff
@@ -0,0 +1,156 @@
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.Objects.requireNonNull;
+
+public interface WithManagerSpec
+{
+    default ManagerSpec managerSpec(List<ResourceGroupSpec> resourceGroupSpec, List<SelectorSpecBuilder> selectors)
+    {
+        return new ManagerSpec(
+                resourceGroupSpec,
+                selectors.stream()
+                        .map(SelectorSpecBuilder::build)
+                        .collect(toImmutableList()),
+                Optional.empty());
+    }
+
+    default ManagerSpec managerSpec(ResourceGroupSpec resourceGroupSpec, List<SelectorSpecBuilder> selectors)
+    {
+        return managerSpec(ImmutableList.of(resourceGroupSpec), selectors);
+    }
+
+    default ResourceGroupSpec resourceGroupSpec(String segmentName)
+    {
+        return new ResourceGroupSpec(
+                new ResourceGroupNameTemplate(segmentName),
+                DataSize.of(100, MEGABYTE).toString(),
+                10,
+                Optional.empty(),
+                Optional.of(10),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty());
+    }
+
+    default ResourceGroupIdTemplate groupIdTemplate(String groupIdTemplate)
+    {
+        return new ResourceGroupIdTemplate(groupIdTemplate);
+    }
+
+    default SelectorSpecBuilder selectorSpec(ResourceGroupIdTemplate groupIdTemplate)
+    {
+        return new SelectorSpecBuilder(
+                new SelectorSpec(
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        groupIdTemplate));
+    }
+
+    class SelectorSpecBuilder
+    {
+        private final SelectorSpec spec;
+
+        private static Optional<Pattern> mergeRegexAndSet(Optional<Pattern> regex, Optional<Set<String>> list)
+        {
+            return mergeTwoRegexes(regex, convertToPattern(list));
+        }
+
+        private static Optional<Pattern> mergeTwoRegexes(Optional<Pattern> first, Optional<Pattern> second)
+        {
+            return Stream.of(first, second)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .map(Pattern::pattern)
+                    .reduce((firstRegexString, secondRegexString) -> String.join("|", firstRegexString, secondRegexString))
+                    .map(Pattern::compile);
+        }
+
+        private static Optional<Pattern> convertToPattern(Optional<Set<String>> list)
```

Comment:
- `Set list` looks weird
- indicate it's not patterns, it's treated as fixed strings
- make this method take List/Set/Collection but not Optional - handle Optional-ness outside

so:

```
private static Pattern convertToPattern(Collection<String> literals)
{
    return Pattern.compile(literals.stream()
            .map(Pattern::quote)
            .collect(joining("|")));
}
```